libraries/core_libs/consensus/include/pbft/round.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "pbft/round_face.hpp"
+#include "pbft/step/certify.hpp"
+#include "pbft/step/filter.hpp"
+#include "pbft/step/finish.hpp"
+#include "pbft/step/polling.hpp"
+#include "pbft/step/propose.hpp"
+#include "pbft/step/step.hpp"
+
+namespace taraxa {
+
+class Round : public RoundFace {
+ public:
+  Round(uint64_t id, std::shared_ptr<NodeFace> node);
+  ~Round();
+
+  static std::shared_ptr<Round> make(uint64_t id, std::optional<uint64_t> step, std::shared_ptr<NodeFace> node);
+
+  void start(std::optional<uint64_t> step = {}) override;
+  void sleepUntil(const time_point& time);
+  void sleepUntil(const std::chrono::milliseconds& time) override;
+  /**
+   * @brief Set PBFT step
+   * @param pbft_step PBFT step
+   */
+  void updateStepData();
+
+  void finish() override;
+
+  void setStep(std::unique_ptr<Step> step) override {
```

Comment:
based on usage, here should be: std::unique_ptr<Step>&& step and method should be private